.github/workflows/tests.yml

```diff
@@ -356,28 +391,92 @@ jobs:
           brew test-bot --only-cleanup-after
           rm -rvf bottles
 
-  test_deps:
-    needs: [tap_syntax, setup_tests, tests]
+  setup_dep_runners:
```

Comment:
I made this a separate job since it takes much longer than determining runners for the `tests` job, so I'd rather not end up having the `tests` job wait for it when the `tests` job doesn't need it.